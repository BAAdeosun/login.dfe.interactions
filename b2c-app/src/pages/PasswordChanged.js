import React from 'react';
import components from '../components';
import { getB2CLink } from '../helpers/urls';
import { ACTIONS } from '../constants/actions';

export default function PasswordChanged() {
    return (
        <div id="passwordChanged">

            <div className="govuk-width-container">
                <components.Breadcrumbs />

                <div id="pageLevelErrorContainer"></div>

                <main className="govuk-main-wrapper">
                    <div className="govuk-grid-row">
                        <div className="govuk-grid-column-two-thirds">
                            <components.PageTitle size='xl' title="We've changed your password"/>
                            <p className="govuk-body">Use your new details to sign in and access your information.</p>
                            <a href={getB2CLink(ACTIONS.LOGIN)} role="button" draggable="false" className="govuk-button govuk-button--start" data-module="govuk-button">
                                Sign in to your account
                            </a>

                            {/* B2C component will be ignored as we don't need anything from it (it has to be loaded though) */}
                            <components.B2C />
                        </div>
                    </div>
                </main>

            </div>

            <script src="__--b2cPath--__/b2c/assets/js-static/pages/passwordChanged.js"></script>

        </div>
    )
}
