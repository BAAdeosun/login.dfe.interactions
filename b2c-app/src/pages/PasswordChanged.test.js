
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PasswordChanged from './PasswordChanged';

it('renders without crashing', () => {
    shallow(<PasswordChanged />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<PasswordChanged />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});