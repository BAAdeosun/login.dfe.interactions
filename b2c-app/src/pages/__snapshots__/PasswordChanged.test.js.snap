// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="passwordChanged"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We've changed your password
            </h1>
            <div>
              <p
                className="govuk-body"
              >
                Use your new details to sign in and access your information.
              </p>
              <a
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                draggable="false"
                id="signInLink"
                onClick={null}
                role="button"
              >
                Sign in to your account
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
